src/main/java/com/linecorp/armeria/server/thrift/ThriftServiceCodec.java

```diff
@@ -559,4 +559,13 @@ protected TProtocol initialValue() {
             return protoFactory.getProtocol(new TByteBufTransport());
         }
     }
+
+    private static Set<Class<?>> getAllInterfaces(Class<?> clz) {
+        Set<Class<?>> interfaces = new HashSet<>();
+        Collections.addAll(interfaces, clz.getInterfaces());
+        if (clz.getSuperclass() != null) {
+            interfaces.addAll(getAllInterfaces(clz.getSuperclass()));
```

Comment:
Craving for `@tailrec` in Java too...

The impl in commons-lang uses a loop: http://commons.apache.org/proper/commons-lang/javadocs/api-release/src-html/org/apache/commons/lang3/ClassUtils.html#line.433